### PR TITLE
GPII-3860: Enable binauth (exekube)

### PR DESCRIPTION
**Pull request by mrtyler**
This PR is an alternate implementation of k8s binauth, replacing #402 (for my testing, I used a much simpler version of `bin_auth.tf` as seen in this PR). Details on why we chose this alternate will be in the ticket (in progress).

This PR is closely related to https://github.com/gpii-ops/gpii-infra/pull/451.

Deploy plan will be in ticket (EDIT: https://issues.gpii.net/browse/GPII-3860?focusedCommentId=39513&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-39513).

### CODE REVIEW

File: modules/gke-cluster/bin_auth.tf [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  # This set of patterns covers Google-provided system images. It would be nice
+  # to use the `globalPolicyEvaluationMode` field in the API
+  # (https://cloud.google.com/binary-authorization/docs/reference/rest/v1/Policy)
+  # but the google provider doesn't support it :(.
+  #
+  # The GENERATED section is manually generated by this process:
+  #
+  # 1. Collect list of Google-provided system images, e.g. the short list at
```
Comment (stepanstipl) [inline modules/gke-cluster/bin_auth.tf]: I would remove the comment from this point below and just reference the GCP doc listing those 4 patterns below (https://cloud.google.com/binary-authorization/docs/configuring-policy-cli#global_policy_evaluation_mode). Reading & following the comment is longer than copying those 4 lines and the comment as a whole distracts from the actual reason - why those 4 patterns.

**Comment by natarajaya** [comment]
LGTM

+1 To Stepan's suggestion to use list instead of bunch of variables.

File: modules/gke-cluster/variables.tf [hunk 2]
```diff
@@ -53,44 +53,9 @@ variable "binary_authorization_enforcement_mode" {
...
-  # This value cannot be empty, so use a placeholder value that satisfies Google's API.
-  default = "PLACE.HOLDER/PATTERN"
-}
-
-variable "binary_authorization_admission_whitelist_pattern_7" {
-  # This value cannot be empty, so use a placeholder value that satisfies Google's API.
-  default = "PLACE.HOLDER/PATTERN"
+variable "binary_authorization_admission_whitelist_patterns" {
+  # This list cannot be empty, so use a placeholder value that satisfies Google's API.
+  default = ["PLACE.HOLDER/PATTERN"]
```
Comment (stepanstipl) [inline modules/gke-cluster/variables.tf]: Is this still true - that the list can't be empty?

Comment (mrtyler) [reply modules/gke-cluster/variables.tf]: Yes.

Comment (stepanstipl) [reply modules/gke-cluster/variables.tf]: Can you elaborate on why please? I don't think I understand.

**Comment by amatas** [comment]
LGTM

File: modules/gke-cluster/variables.tf (hunk 2 above)
Comment (mrtyler) [reply modules/gke-cluster/variables.tf]: I added the error message to the comment.

**Pull request merged**